### ISSUE TITLE
`nstreams` used before being initialized (compiling with `-init=huge`)

### ISSUE THREAD

**Issue by phil-blain** [phil-blain/CICE#43]
`nstreams` is initialized in `ice_history::init_hist`, but is used in `ice_calendar::calendar`, called from `ice_calendar::init_calendar`. The calling sequence in `CICE_InitMod::cice_init` is:

~~~fortran
132           call init_calendar        ! initialize some calendar stuff
133           call init_hist (dt)       ! initialize output history file
~~~

So it's used before being initialized. This was found by compiling with `-init=huge` which initializes integers to their maximum value; `nstreams` was 2147483647 in the core dump and the model aborted because `histfreq` was accessed at index 6 while it has size 5 (thanks to `-check`).

Might be worth it to run the full suite with that flag, see what we uncover!

**Comment by phil-blain** [comment]
So it seems `-init=huge,arrays` still initializes reals to NaN, so it's hard to see if we uncover new bugs just due to `huge`. I'll try with just `huge`.

**Comment by phil-blain** [comment]
No new failures with just `-init=huge`. I've fixed the above in 584f370 and I'll open a PR for that.

**Comment by phil-blain** [comment]
PR merged: https://github.com/CICE-Consortium/CICE/pull/752
